### ISSUE TITLE
Order for the puzzles or how were the puzzles curated

### ISSUE THREAD

**Issue by ArmenJeddi** [niladridutt/monetGPT#6]
I'm super curious about how you came up with the current puzzles and whether there were other puzzles you tried in this process.

Another question is, when you train the model over the puzzles, is there an order to it? e.g. do you first train only on puzzle 1 and then puzzle 2, etc.?  I mean is there some sort of curriculum learning in play? moreover, do you have any metrics that measure how well the model does on the puzzles themselves?

**Comment by niladridutt** [comment]
We tried to follow human logic and learning when creating the puzzles. I think there can be lots of more variations with the puzzles. We did not try other ones but it's very likely that there are better puzzles to try!

We shuffled up the training to avoid overfitting to a single puzzle type. There could in theory be some curriculum for example when generating the reasoning. We use a pretrained LLM for this but one can try to generate the reasoning for puzzle 2 after training it on puzzle 1. 

No metrics sorry :(

Thanks for the insightful questions!
